sklearn/tests/test_calibration.py

```diff
@@ -0,0 +1,180 @@
+                                       sparse.csr_matrix(X_test))]:
+        for method in ['isotonic', 'sigmoid']:
+            pc_clf = CalibratedClassifierCV(clf, method=method, cv=2)
+            # Note that this fit overwrites the fit on the entire training
+            # set
+            pc_clf.fit(this_X_train, y_train, sample_weight=sw_train)
+            prob_pos_pc_clf = pc_clf.predict_proba(this_X_test)[:, 1]
+
+            # Check that brier score has improved after calibration
+            assert_true(brier_score_loss(y_test, prob_pos_clf) >
+                        brier_score_loss(y_test, prob_pos_pc_clf))
+
+            # Check invariance against relabeling [0, 1] -> [1, 2]
+            pc_clf.fit(this_X_train, y_train + 1, sample_weight=sw_train)
+            prob_pos_pc_clf_relabeled = pc_clf.predict_proba(this_X_test)[:, 1]
+            assert_array_almost_equal(prob_pos_pc_clf,
+                                      prob_pos_pc_clf_relabeled)
+
+            # Check invariance against relabeling [0, 1] -> [-1, 1]
+            pc_clf.fit(this_X_train, 2 * y_train - 1, sample_weight=sw_train)
+            prob_pos_pc_clf_relabeled = pc_clf.predict_proba(this_X_test)[:, 1]
+            assert_array_almost_equal(prob_pos_pc_clf,
+                                      prob_pos_pc_clf_relabeled)
+
+            # Check invariance against relabeling [0, 1] -> [1, 0]
+            pc_clf.fit(this_X_train, (y_train + 1) % 2,
+                       sample_weight=sw_train)
+            prob_pos_pc_clf_relabeled = \
+                pc_clf.predict_proba(this_X_test)[:, 1]
+            if method == "sigmoid":
+                assert_array_almost_equal(prob_pos_pc_clf,
+                                          1 - prob_pos_pc_clf_relabeled)
+            else:
+                # Isotonic calibration is not invariant against relabeling
+                # but should improve in both cases
+                assert_true(brier_score_loss(y_test, prob_pos_clf) >
+                            brier_score_loss((y_test + 1) % 2,
+                                             prob_pos_pc_clf_relabeled))
+
+
+def test_calibration_multiclass():
+    """Test calibration for multiclass """
+    # test multi-class setting with classifier that implements
+    # only decision function
+    clf = LinearSVC()
+    X, y = make_blobs(n_samples=100, n_features=2, random_state=42,
+                      cluster_std=3.0)
+    X_train, y_train = X[::2], y[::2]
+    X_test, y_test = X[1::2], y[1::2]
+    for method in ['isotonic', 'sigmoid']:
+        cal_clf = CalibratedClassifierCV(clf, method=method, cv=2)
+        cal_clf.fit(X_train, y_train)
+        probas = cal_clf.predict_proba(X_test)
+        assert_array_almost_equal(np.sum(probas, axis=1), np.ones(len(X_test)))
+        assert_true(clf.fit(X_train, y_train).score(X_test, y_test) <=
+                    cal_clf.fit(X_train, y_train).score(X_test, y_test))
```

Comment:
Actually, I am not sure that this assertion makes sense in the first place. Why would probability scaling improve the hard prediction score in the first place? The `score` method of classifiers returns the accuracy (that is the opposite of the zero-one loss). AFAIK it should not necessarily be improved by probability calibration: the location of the decision threshold might move a bit but I see no principled reason why probability would necessarily improve the exact location of the decision boundary w.r.t. the accuracy score.